### DISCUSSION TITLE
Reusing rendered Pandas dataframe as table in article?

### DISCUSSION THREAD

**Discussion by pschumm** [jupyter-book/mystmd#1067] [Q&A] [answered]
I've been getting acquainted with `mystmd` and am really enjoying it! Just one thing that seems to be missing. The documentation clearly shows how to reuse a figure generated in a notebook with something like:

```
:::{figure} #my-cell
:name: fig-my-cell
:::
```

Is it possible to do the same with a table? Specifically, can I reuse a table such as that created by [pandas.crosstab()](https://pandas.pydata.org/docs/reference/api/pandas.crosstab.html) (i.e., a dataframe rendered as HTML in a notebook cell's output) within an article, without having to recreate the table in Markdown?

Perhaps I'm just missing something obvious here? Thanks very much in advance for any pointers!

**Comment by rowanc1** [comment, marked as answer]
Thanks @pschumm. You can add a `:kind: table` to the figure, and it will work. In addition to that you can wrap it in a `table` directive and use an `embed` or `![](#pd-table)` inside of the body of the table directive. These will give you the `source` and the table caption.

![image](https://github.com/executablebooks/meta/assets/913249/ae971e8a-ea91-4111-85d2-855e9c10e288)

```
:::{table} This is my table
:label: tbl:here
![](#pd-table)
:::

:::{figure} #pd-table
:label: tbl:there
:kind: table
This is my table caption
:::
```

Over in jupyter, the cell needs to be labeled with `#| label: pd-table`.

Hope that helps!

**Comment by pschumm** [comment]
Sorry for asking in two places—I mistakenly asked first in `executablebooks`, but then realized it was probably more appropriate to do so here. Won't make that mistake again.
